Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -299,19 +299,13 @@ public void OnPoolMemberException(IPooledConnection connection, Exception except
     {
         if (exception is TokenExpiredException)
         {
-            // if a token exception occured the pool member, all connections in the pool that are using that token
-            // should be closed. This is because the token is now invalid and all connections using it will fail.
-            foreach (var conn in _inUseConnections)
-            {
-                if (connection.AuthToken.Equals(conn.AuthToken))
-                {
-                    conn.AuthorizationStatus = AuthorizationStatus.TokenExpired;
-                }
-            }
+            connection.AuthorizationStatus = AuthorizationStatus.SecurityError;
         }
 
         if (exception is AuthorizationException)
         {
+            // if an auth exception occured the pool member, all connections in the pool that are using that token
+            // should be closed. This is because the token is now invalid and all connections using it will fail.
             foreach (var conn in _inUseConnections)
```

Comment:
do we care about setting it for idle connections, _inUseConnections are connections that another session or transaction currently has borrowed.

Reply:
why do we want idle connections to lazily discover the auth token is expired but set in use connections as expired?

Reply:
Yeah, thinking about it, you're right, let me alter that.